### ISSUE TITLE
[Notifier] Support the new Firebase HTTP API

### ISSUE THREAD

**Issue by acasademont** [symfony/symfony#36549]
**Description**  
In #33968 preliminary support for Firebase notifications was added. However, that HTTP API is currently marked as legacy by Google (https://firebase.google.com/docs/cloud-messaging/http-server-ref) and it's encouraged to migrate to the new API (https://firebase.google.com/docs/cloud-messaging/migrate-v1). Maybe it's now the time to do it before 5.1 is released? The component is still marked as experimental, so I guess BC-breaks are acceptable?

Basically the new API is much more convenient when sending a notification to multiple device platforms (iOS, Android) at once. On the old one, you needed to target a specific device platform.

**Comment by stof** [comment]
Using the new API would indeed be better. Can you send a pull request contributing it ?

**Comment by acasademont** [comment]
Sure, will try to work on it this weekend

**Comment by acasademont** [comment]
Took a look at this and it's seems quite difficult to properly implement a Firebase Notifier due to the complicated OAuth flow that Google requires. You first need to ask for an OAuth token using your service account with the requested scopes and only then you can access the Firebase endpoint. This is quite incompatible with the the current DSN flow with a username and password/key...

I'm not even sure the already merged Firebase Notifier does work, the old API also needed this oauth token and right now the token is just created concatenating a username and a password (https://github.com/symfony/symfony/blob/4170346ea736d90de32384cf4fea45f2aa0f4d73/src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseTransportFactory.php#L29).

**Comment by fabpot** [comment]
If that's the case, we will need to revert it to avoid keeping something that does not work. Can you confirm?

**Comment by acasademont** [comment]
I'm not using the google legacy API in production so I can't really confirm with 100% certainty. I investigated a bit more and it could work, the legacy API does not use the new OAuth flow, but not in the current form.

In the legacy API (https://firebase.google.com/docs/cloud-messaging/auth-server#authorize-legacy-protocol-send-requests) you have server key that needs to be set as a header `Authorization:key=AIzaSyZ-1u...0GBYzPu7Udno5aA` But in the current form, the `FirebaseTransportFactory` expects both a user and a password in the DSN that will be concatenated with a ":" to form the key. To be able to make it work it should only expect a user, with the user being the server key listed in your Firebase console.

Maybe @Jeroeny could give us a bit more insight.

**Comment by Jeroeny** [comment]
@acasademont Hi, this is because that server token has a `:` in it. Therefor in the DSN it is seen as a "user:password", and needs concatenation to token in the factory. It does work though, I've tested it locally and was able to receive notifications. If you can use help with the new API let me know.

**Comment by acasademont** [comment]
Ah! That makes sense then, nice trick :p 

Closing this then, with the new Oauth Flow, it's pretty impossible to implement the new API.
